eclass/user.eclass

```diff
@@ -130,19 +145,22 @@ enewuser() {
 	if [[ -n ${euid} && ${euid} != -1 ]] ; then
 		if [[ ${euid} -gt 0 ]] ; then
 			if [[ -n $(egetent passwd ${euid}) ]] ; then
+				[[ -n ${force_uid} ]] && die "${FUNCNAME}: UID ${euid} already taken"
 				euid="next"
 			fi
 		else
 			eerror "Userid given but is not greater than 0 !"
 			die "${euid} is not a valid UID"
 		fi
 	else
+		[[ -n ${force_uid} ]] && die "${FUNCNAME}: -F with uid==-1 makes no sense"
 		euid="next"
 	fi
 	if [[ ${euid} == "next" ]] ; then
-		for ((euid = 101; euid <= 999; euid++)); do
+		for ((euid = 101; euid <= 1000; euid++)); do
 			[[ -z $(egetent passwd ${euid}) ]] && break
 		done
+		[[ ${euid} -eq 1000 ]] && die "${FUNCNAME}: no free UID found"
```

Comment:
When the for loop terminates without success, the euid counter will have a value of 1001:
`for ((euid = 101; euid <= 1000; euid++)); do false; done; echo ${euid}`
`1001`
So the previous <= 999 as condition to terminate the loop was correct.

Also, purely cosmetic, but I'd revert the test, to make it clear that it's the same condition (and `|| die` looks nicer). Like this:
`[[ ${euid} -le 999 ]] || die "${FUNCNAME}: no free UID found"`

Reply:
Done.